### PR TITLE
Kernel: Try to reuse empty slabheaps before expanding the kmalloc-heap

### PR DESCRIPTION
**Pull request by Hendiadyoin1**

### CODE REVIEW

File: Kernel/Heap/kmalloc.cpp [hunk 1]
```diff
@@ -167,6 +167,29 @@ class KmallocSlabheap {
         return total;
     }
 
+    bool try_purge()
+    {
+        bool did_purge = false;
+
+        // FIXME: We cannot remove children from the list we are iterating through a structured loop,
+        //        so we have do iterate manually
```
Comment (kleinesfilmroellchen) [inline Kernel/Heap/kmalloc.cpp:175]: This can be a simple NOTE, nothing we can do to improve it.

File: Kernel/Heap/kmalloc.cpp [hunk 2]
```diff
@@ -204,6 +227,22 @@ struct KmallocGlobalData {
...
         }
 
+        // NOTE: This size calculation is a mirror of kmalloc_aligned(KmallocSlabBlock)
+        if (size <= KmallocSlabBlock::block_size * 2 + sizeof(ptrdiff_t) + sizeof(size_t)) {
+            // FIXME: We should propagate a freed pointer, to find the specific subheap it belonged to
+            //        This would save us iterating over them in the next step and remove a recursion
+            bool did_purge = false;
+            for (auto& slabheap : slabheaps) {
+                if (slabheap.try_purge()) {
+                    dbgln_if(KMALLOC_DEBUG, "Kmalloc purged block(s) from slabheap of size {} to avoid expansion", slabheap.slab_size());
```
Comment (kleinesfilmroellchen) [inline Kernel/Heap/kmalloc.cpp:237]: Can this not be a standard dbgln, like heap expansion, or is this very spammy?

Comment (Hendiadyoin1) [reply Kernel/Heap/kmalloc.cpp:237]: Heap expansion is also a dbgln_if, that's why I chose this here, too

File: Kernel/Heap/kmalloc.cpp (hunk 1 above)
Comment (Hendiadyoin1) [reply Kernel/Heap/kmalloc.cpp:175]: Also reworded it a bit

**Pull request merged**